doc-processor-gradle-plugin/src/main/kotlin/nl/jolanrensen/docProcessor/defaultProcessors/IncludeArgDocProcessor.kt

```diff
@@ -0,0 +1,242 @@
+        if (i > 0 && !anyModifications) {
+            val argsNotFound = argsNotFound.flatMap { (documentable, args) ->
+                args.map { arg -> "`${documentable.path}` -> @$useArgumentTag $arg" }
+            }.joinToString("\n")
+            println("IncludeArgDocProcessor WARNING: Could not find all arguments:[\n$argsNotFound\n]")
+            return false
+        }
+
+        return super.shouldContinue(i, anyModifications, parameters, filteredDocumentables, allDocumentables)
+    }
+
+    // @arg map for path -> arg name -> value
+    private val argMap: MutableMap<DocumentableWrapper, MutableMap<String, String>> = mutableMapOf()
+
+    private val argsNotFound: MutableMap<DocumentableWrapper, MutableSet<String>> = mutableMapOf()
+
+    private fun process(
+        tagWithContent: String,
+        documentable: DocumentableWrapper,
+        allDocumentables: Map<String, List<DocumentableWrapper>>,
+    ): String {
+        val tagName = tagWithContent.getTagNameOrNull()
+        val isArgDeclaration = tagName == declareArgumentTag
+
+        val argTagsStillPresent = declareArgumentTag in documentable.docContent.findTagNamesInDocContent()
+
+        return when {
+            isArgDeclaration -> { // @arg
+                val argArguments = tagWithContent.getTagArguments(
+                    tag = declareArgumentTag,
+                    numberOfArguments = 2,
+                )
+                val originalKey = argArguments.first()
+                var keys = listOf(originalKey)
+
+                if (originalKey.startsWith('[') && originalKey.contains(']')) {
+                    keys = buildReferenceKeys(originalKey, documentable, allDocumentables)
+                }
+
+                val value = argArguments.getOrElse(1) { "" }
+                    .trimStart()
+                    .removeSuffix("\n")
+                    .removeEscapeCharacters()
+
+                argMap.getOrPut(documentable) { mutableMapOf() }.apply {
+                    for (it in keys) this[it] = value
+                }
+
+                argsNotFound.getOrPut(documentable) { mutableSetOf() } -= keys.toSet()
+                ""
+            }
+
+            argTagsStillPresent -> {
+                // skip @includeArg tags if there are still @args present
+                tagWithContent
+            }
+
+            else -> { // @includeArg
+                val includeArgArguments = tagWithContent.getTagArguments(
+                    tag = useArgumentTag,
+                    numberOfArguments = 2,
+                )
+
+                val originalKey = includeArgArguments.first()
+
+                // for stuff written after the @includeArg tag, save and include it later
+                val extraContent = includeArgArguments.getOrElse(1) { "" }
+                    .trimStart()
+
+                var keys = listOf(originalKey)
+
+                // TODO?: Java {@link ReferenceLinks}
+                if (javaLinkRegex in originalKey) {
```

Comment:
again, it's hard to mix Java and Kotlin docs support, we need to better isolate both paths in the code and don't check, say, javaRegex in the keys

Reply:
From my perspective, it looks like a weak point in code and architecture, but it's not a blocker for us

Reply:
I'll strengthen it with a `documentable.programmingLanguage == JAVA`, so it won't trigger in Kotlin code at all. I don't want a large split between kdoc and javadoc since they are so similar. Most processors don't mind which language is targeted and if they do they can make a difference based on `documentable.programmingLanguage`